File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        return search(nums,target);

    }
    public static int search(int []nums,int target){
        int left=0;
        int right=nums.length-1;
        while(left<=right){
            int mid=left+(right-left)/2;
            if(target==nums[mid]){
                return mid;
            }if(target>nums[mid]){
                left=mid+1;
            }else{
                right=mid-1;
            }
            
        }
        return left;
    }
}

File: 35-search-insert-position/search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l=0
        r=len(nums)-1
        while(l<=r):
            mid=l+(r-l)/2
            if(nums[mid]==target):
                return mid
            if(nums[mid]>target):
                r=mid-1
            else:
                l=mid+1
        return l
        